src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,41 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int FIGURES_TYPES_QUANTITY = 5;
+    private static final int MAX_RANDOM_OF_FIGURES_PARAMETERS = 100;
+    private static final String DEFAULT_COLOR = Colors.WHITE.name().toLowerCase();
+    private static final int DEFAULT_RADIUS = 10;
+    private final Random random = new Random();
+
+    public Figure getRandomFigure() {
+        int index = random.nextInt(FIGURES_TYPES_QUANTITY) + 1;
+        String randomColor = new ColorSupplier().getRandomColor();
+        switch (index) {
+            case 1:
+                return new Circle(randomColor,
+                        random.nextInt(MAX_RANDOM_OF_FIGURES_PARAMETERS + 1));
```

Comment:
Not fixed:

Don't pass expressions in the constructor, instead create variables. See [here](https://mate-academy.github.io/jv-program-common-mistakes/java-core/abstract-class-interface/oop-advanced#%D1%81reating-a-figure-dont-pass-expressions-in-constructor).

Applicable to all other cases.

```suggestion
                int radius = random.nextInt(MAX_RANDOM_OF_FIGURES_PARAMETERS + 1);
                return new Circle(randomColor, radius);
```